spring-integration-core/src/main/java/org/springframework/integration/endpoint/SourcePollingChannelAdapter.java

```diff
@@ -77,12 +77,30 @@ public void setOutputChannel(MessageChannel outputChannel) {
 		this.outputChannel = outputChannel;
 	}
 
+	/**
+	 * Return this endpoint's source.
+	 * @return the source.
+	 * @since 4.3
+	 */
+	public MessageSource<?> getMessageSource() {
+		return this.source;
+	}
+
 	public void setOutputChannelName(String outputChannelName) {
 		Assert.hasText(outputChannelName, "'outputChannelName' must not be empty");
 		this.outputChannelName = outputChannelName;
 	}
 
 	/**
+	 * Return the name of the output channel.
+	 * @return the output channel name.
+	 */
+	public String getOutputChannelName() {
+		return this.outputChannel == null ? this.outputChannelName : this.outputChannel instanceof NamedComponent ?
+				((NamedComponent) this.outputChannel).getComponentName() : null;
```

Comment:
This method is fully redundant. We can use `getOutputChannel().toString()` from the `GraphServer`.
And yes: the result of `getOutputChannel()` cannot be `null`. See `onInit()` in this class,